AIOps-OpniPreProcessor/src/main/java/org/opensearch/opnipreprocessing/plugin/OpniPreProcessor.java

```diff
@@ -273,19 +274,38 @@ else if (ingestDocument.hasField("COMM")){
                 if (kubernetes.containsKey("container_image") && ((String)kubernetes.get("container_image")).contains("longhornio-")) {
                     logType = "longhorn";
                 }
-                ingestDocument.setFieldValue("pod_name", ((String)kubernetes.get("pod_name")));
+                if (kubernetes.containsKey("pod_name")) {
+                    podName = ((String)kubernetes.get("pod_name"));
+                }
+                if (kubernetes.containsKey("namespace_name")) {
+                    namespaceName = ((String)kubernetes.get("namespace_name"));
+                } 
             }        
-        }  
+        }
+        if (ingestDocument.hasField("deployment")) {
+            deployment = ingestDocument.getFieldValue("deployment", String.class);
+        }
+        if (ingestDocument.hasField("service")) {
+            service = ingestDocument.getFieldValue("service", String.class);
+        }    
         ingestDocument.setFieldValue("log_type", logType);
         ingestDocument.setFieldValue("kubernetes_component", kubernetesComponent);
+        ingestDocument.setFieldValue("pod_name", podName);
```

Comment:
should we set those values for workload logs only?

Reply:
We do have Longhorn and Rancher logs which do have pods and deployment data so might be worthwhile to keep those for anything we do down the road.

Reply:
longhorn and rancher logs have their own pertained models

Reply:
They do but if we want to do further analysis down the road where for Rancher and Longhorn log anomaly detection, we want to correlate it with a particular pod and deployment, then it'd be good to have that data available.

Reply:
OK. and why do we need set those fields as empty string if not exist?

Reply:
and why do you get  `deployment` and `service` and set the same values back?